pkg/addon/addon.go

```diff
@@ -418,15 +424,74 @@ func createGitHelper(content *utils.Content, token string) *gitHelper {
 	}
 }
 
+func createGiteeHelper(content *utils.Content, token string) *giteeHelper {
+	var ts oauth2.TokenSource
+	if token != "" {
+		ts = oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	}
+	tc := oauth2.NewClient(context.Background(), ts)
+	tc.Timeout = time.Second * 20
+	cli := NewGiteeClient(tc, nil)
+	return &giteeHelper{
+		Client: cli,
+		Meta:   content,
+	}
+}
+
 // readRepo will read relative path (relative to Meta.Path)
 func (h *gitHelper) readRepo(relativePath string) (*github.RepositoryContent, []*github.RepositoryContent, error) {
-	file, items, _, err := h.Client.Repositories.GetContents(context.Background(), h.Meta.Owner, h.Meta.Repo, path.Join(h.Meta.Path, relativePath), nil)
+	file, items, _, err := h.Client.Repositories.GetContents(context.Background(), h.Meta.GithubContent.Owner, h.Meta.GithubContent.Repo, path.Join(h.Meta.GithubContent.Path, relativePath), nil)
 	if err != nil {
 		return nil, nil, WrapErrRateLimit(err)
 	}
 	return file, items, nil
 }
 
+// readRepo will read relative path (relative to Meta.Path)
+func (h *giteeHelper) readRepo(relativePath string) (*github.RepositoryContent, []*github.RepositoryContent, error) {
+	file, items, err := h.Client.GetGiteeContents(context.Background(), h.Meta.GiteeContent.Owner, h.Meta.GiteeContent.Repo, path.Join(h.Meta.GiteeContent.Path, relativePath), h.Meta.GiteeContent.Ref)
+	if err != nil {
+		return nil, nil, WrapErrRateLimit(err)
+	}
+	return file, items, nil
+}
+
+// GetGiteeContents can return either the metadata and content of a single file
+func (c *Client) GetGiteeContents(ctx context.Context, owner, repo, path, ref string) (fileContent *github.RepositoryContent, directoryContent []*github.RepositoryContent, err error) {
+	escapedPath := (&url.URL{Path: path}).String()
+	u := fmt.Sprintf(c.BaseURL.String()+"repos/%s/%s/contents/%s", owner, repo, escapedPath)
+	if ref != "" {
+		u = fmt.Sprintf(u+"?ref=%s", ref)
+	}
+
+	req, err := http.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	response, err := c.Client.Do(req.WithContext(ctx))
+	defer func() {
+		if err := response.Body.Close(); err != nil {
+			fmt.Println(err)
+		}
+	}()
+	if err != nil {
+		return nil, nil, err
+	}
+	content, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, nil, err
+	}
+	fileUnmarshalError := json.Unmarshal(content, &fileContent)
+	if fileUnmarshalError == nil {
+		return fileContent, nil, nil
+	}
+	directoryUnmarshalError := json.Unmarshal(content, &directoryContent)
+	if directoryUnmarshalError == nil {
+		return nil, directoryContent, nil
+	}
```

Comment:
maybe it's better to make these lines in a single function, and write some unit tests for it.